Tests/FuzzilliTests/LifterTest.swift

```diff
@@ -363,6 +363,50 @@ class LifterTests: XCTestCase {
 
         XCTAssertEqual(lifted_program,expected_program)
     }
+
+    func testTryCatchFinallyLifting(){
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+
+        let f = b.definePlainFunction(withSignature: FunctionSignature(withParameterCount: 3)) { args in
+            b.beginTry() {
+                let v = b.binary(args[0], args[1], with: .Mul)
+                b.doReturn(value: v)
+            }
+            b.beginCatch() { _ in
+                let v4 = b.createObject(with: ["a" : b.loadInt(1337)])
+                b.reassign(args[0], to: v4)
+            }
+            b.beginFinally() {
+                let v = b.binary(args[0], args[1], with: .Add)
+                b.doReturn(value: v)
+            }
+            b.endTryCatch()
+        }
+        b.callFunction(f, withArgs: [b.loadBool(true), b.loadInt(1)])
+
+        let program = b.finalize()
+
+        let lifted_program = fuzzer.lifter.lift(program)
+        let expected_program = """
+        function v0(v1,v2,v3) {
+            try {
```

Comment:
Maybe also include a try-catch and a try-finally statement (within the same program or in another one)?

Reply:
testcases updated